Source/WebCore/html/HTMLInputElement.cpp

```diff
@@ -2103,6 +2103,25 @@ bool HTMLInputElement::isEmptyValue() const
     return m_inputType->isEmptyValue();
 }
 
+bool HTMLInputElement::isDevolvableWidget() const
+{
+    return m_inputType->isTextType()
+        || m_inputType->isTelephoneField()
+        || m_inputType->isURLField()
+        || m_inputType->isURLField()
+        || m_inputType->isEmailField()
+        || m_inputType->isSearchField()
+        || m_inputType->isPasswordField()
+        || m_inputType->isDateField()
+        || m_inputType->isMonthField()
```

Comment:
Not at a laptop right now, but I think there might be a better way to do this, which groups related input types.